registries/adapters/local_openshift_adapter.go

```diff
@@ -74,71 +99,87 @@ func (r LocalOpenShiftAdapter) FetchSpecs(imageNames []string) ([]*bundle.Spec,
 		return nil, err
 	}
 
-	fqImages, err := openshiftClient.ConvertRegistryImagesToSpecs(imageNames)
+	listImages, err := openshiftClient.ListRegistryImages()
 	if err != nil {
 		log.Errorf("Failed to load registry images")
 		return nil, err
 	}
 
-	for _, image := range fqImages {
-		spec, err := r.loadSpec(image.DecodedSpec)
-		if err != nil {
-			log.Errorf("Failed to load image spec")
-			continue
-		}
-		spec.Runtime, err = getAPBRuntimeVersion(image.Runtime)
-		if err != nil {
-			log.Errorf("Failed to parse image runtime version")
-			continue
-		}
-		if strings.HasPrefix(image.Name, registryIP) == false {
-			log.Debugf("Image does not have a registry IP as prefix. This might cause problems but not erroring out.")
-		}
-		if r.Config.Namespaces == nil {
-			log.Debugf("Namespace not set. Assuming `openshift`")
-			r.Config.Namespaces = append(r.Config.Namespaces, "openshift")
-		}
-		spec.Image = image.Name
-		nsList := strings.Split(image.Name, "/")
-		var namespace string
-		if len(nsList) == 0 {
-			log.Errorf("Image [%v] is not in the proper format. Erroring.", image.Name)
-			continue
-		} else if len(nsList) < 3 {
-			// Image does not have any registry prefix. May be a product of S2I
-			// Expecting openshift/foo-bundle
-			namespace = nsList[0]
-		} else {
-			// Expecting format: 172.30.1.1:5000/openshift/foo-bundle
-			namespace = nsList[1]
-		}
-		for _, ns := range r.Config.Namespaces {
-			// logging to warn users about the potential bug if
-			// the svc-acct does not have access to the namespace.
-			if ns != "openshift" {
-				log.Warningf("You may not be able to load provision images from the namespace: %v.\n"+
-					"You should make sure that the namespace has given the permissions for the "+
-					"system:authenticated group.", ns)
-			}
-			if ns == namespace {
-				log.Debugf("Image [%v] is in configured namespace [%v]. Adding to SpecList.", image.Name, ns)
-				specList = append(specList, spec)
+	for _, image := range listImages.Items {
+		n := strings.Split(image.DockerImageManifest, "@")[0]
+		for _, providedImage := range imageNames {
+			if providedImage == n {
+				spec, err := r.loadSpec(image)
+				if err != nil {
+					log.Errorf("Failed to load image spec")
```

Comment:
In cases like this , as this is a lib, do we want to use something like https://github.com/hashicorp/go-multierror or just implement something similar. Then hand back that set of errors to the consumer?

Reply:
If we did something like multi error, perhaps then we could group this logic into a  ```validateImage(image)error``` where the error here would be a multi error. Again just a general comment not really to do with the openshift client change

Reply:
@maleck13 I think if we can add this comment to the pkg/errors issue above and discuss on Tuesday? I think it is time to start thinking about the UX of the library.

Reply:
I will add it to the issue